js/globalize.js

```diff
@@ -65,18 +65,29 @@ glModule.provider('globalizeWrapper', function () {
             }
 
             mainLoaded = false;
-            loadResources(
-                cldrBasePath + '/main/' + currentLocale,
-                mainResources,
-                function (results) {
-                    mainData = [];
-                    for (var i = 0; i < results.length; i++)
-                        mainData.push(results[i].data);
-                    mainLoaded = true;
-                    finishLoading();
-                }
-            );
+            $http.get(cldrBasePath + '/availableLocales.json').then(function(results) {
+              var availableLocales = results.data.availableLocales;
+
+              var allResources = [];
+              for(var i = 0; i < availableLocales.length; i++) {
+                var locale = availableLocales[i];
+                allResources = allResources.concat(mainResources.map(function(mr) {
+                  return '/'+locale+mr;
+                }));
+              }
+              loadResources(
+                  cldrBasePath + '/main/',
+                  allResources,
+                  function (results) {
+                      mainData = [];
+                      for (var i = 0; i < results.length; i++)
```

Comment:
I dont want to change old code that is not mine :) I am not refactoring but fixing.